src/components/input/code/templates/inputTemplate.jsx

```diff
@@ -0,0 +1,48 @@
+var React = require('react');
+var classNames = require('classnames');
+
+module.exports = function (component){
+
+  var classes = classNames('input-group', {
+    'input-group-error': component.state.error || false,
+    'input-group-disabled': component.props.disabled || false
+  });
+
+  // the form label
+  var label;
+
+  if (component.props.label){
+    label = ( <label className="input-group-label" for={component.props.name}>{component.props.label}</label > );
+  }
+
+  /**
+   * Create the span element used for containing messages
+   * related to the element.
+   */
+  var span;
+
+  if (component.state.error)
+  {
+    span = ( <span className="input-group-span">{component.state.error}</span> );
+  }
+
+  return (
+    <div className={classes} ref={component.props.ref}>
+      {label}
+      <input
+        className="input-group-field"
+        type={component.props.type}
+        valid={component.state.valid}
+        name={component.props.name}
+        id={component.props.id}
+        placeholder={component.props.placeholder}
+        onChange={component.changeHandler}
```

Comment:
You're using the `changeHandler` function that you've defined within the component directly here, is it supposed to be the `onChange` function passed in via props?

You define `onChange` within the `propTypes` object and assign `this.changeHandler` to `onChange` in `getDefaultProps`.